python/lsst/drp/tasks/update_visit_summary.py

```diff
@@ -0,0 +1,755 @@
+                "statistics for the final backgrounds that can be used as-is."
+            ),
+            "replacement": {
+                "The 'background_originals' connection refers to a background "
+                "model that has been superseded by the model referred to by "
+                "the 'background_overrides' connection."
+            },
+            # Could also imagine an option in which there is no original
+            # background and the new one stands alone; can add later if needed.
+        },
+        default="input_summary",
+        optional=False,
+    )
+    # Could imagine an option here to say that the original background has not
+    # been subtracted from the input exposures, allowing postISRCCD to be used
+    # as input exposures.  Can add later if needed.
+
+
+class UpdateVisitSummaryTask(PipelineTask):
+    """A pipeline task that creates a new visit-summary table after all
+    `lsst.afw.image.Exposure` components have been finalized.
+
+    Notes
+    -----
+    This task is designed to be run just prior to making warps for coaddition,
+    as it aggregates all inputs other than the images and backgrounds into a
+    single ``ExposureCatalog`` dataset and recomputes summary statistics that
+    are useful in selecting which images should go into a coadd.  Its output
+    can also be used to reconstruct a final processed visit image when combined
+    with a post-ISR image, the background model, and the final mask.
+    """
+
+    # The `run` method of this task can conditionally apply overrides for PSFs
+    # and aperture corrections, but its `PipelineTask` interface always applies
+    # them.  We can always add the config options to make them optional later,
+    # if that turns out to be useful.
+
+    _DefaultName = "updateVisitSummary"
+    ConfigClass = UpdateVisitSummaryConfig
+
+    compute_summary_stats: ComputeExposureSummaryStatsTask
+
+    def __init__(self, *, initInputs: dict[str, Any] | None = None, **kwargs: Any):
+        super().__init__(initInputs=initInputs, **kwargs)
+        self.makeSubtask("compute_summary_stats")
+        if initInputs is None or "input_summary_schema" not in initInputs:
+            raise RuntimeError("Task requires 'input_summary_schema' in initInputs.")
+        input_summary_schema = initInputs["input_summary_schema"].schema
+        self.schema_mapper = SchemaMapper(input_summary_schema)
+        self.schema_mapper.addMinimalSchema(input_summary_schema)
+        self.schema = self.schema_mapper.getOutputSchema()
+        if self.config.wcs_provider == "tract":
+            self.schema.addField(
+                "wcsTractId", type="L", doc="ID of the tract that provided the WCS."
+            )
+        if self.config.photo_calib_provider == "tract":
+            self.schema.addField(
+                "photoCalibTractId",
+                type="L",
+                doc="ID of the tract that provided the PhotoCalib.",
+            )
```

Comment:
Even though it's not being run automatically here (yet?), I'm just using black here since it's a new file and hence I can, and I'm going to just let it make its choices.